secant/Dependencies/DerivationTool/DerivationToolLiveKey.swift

```diff
@@ -13,51 +13,21 @@ extension DerivationToolClient: DependencyKey {
+            deriveUnifiedViewingKeyFromSpendingKey: { spendingKey in
+                try derivationTool.deriveUnifiedFullViewingKey(from: spendingKey)
             },
-            deriveViewingKey: { spendingKey in
-                try derivationTool.deriveViewingKey(spendingKey: spendingKey)
-            },
-            deriveSpendingKeys: { seed, numberOfAccounts in
-                try derivationTool.deriveSpendingKeys(seed: seed, numberOfAccounts: numberOfAccounts)
-            },
-            deriveShieldedAddress: { seed, accountIndex in
-                try derivationTool.deriveShieldedAddress(seed: seed, accountIndex: accountIndex)
-            },
-            deriveShieldedAddressFromViewingKey: { viewingKey in
-                try derivationTool.deriveShieldedAddress(viewingKey: viewingKey)
-            },
-            deriveTransparentAddress: { seed, account, index in
-                try derivationTool.deriveTransparentAddress(seed: seed, account: account, index: index)
-            },
-            deriveUnifiedViewingKeysFromSeed: { seed, numberOfAccounts in
-                try derivationTool.deriveUnifiedViewingKeysFromSeed(seed, numberOfAccounts: numberOfAccounts)
-            },
-            deriveUnifiedAddressFromUnifiedViewingKey: { uvk in
-                try derivationTool.deriveUnifiedAddressFromUnifiedViewingKey(uvk)
-            },
-            deriveTransparentAddressFromPublicKey: { pubkey in
-                try derivationTool.deriveTransparentAddressFromPublicKey(pubkey)
-            },
-            deriveTransparentPrivateKey: { seed, account, index in
-                try derivationTool.deriveTransparentPrivateKey(seed: seed, account: account, index: index)
-            },
-            deriveTransparentAddressFromPrivateKey: { tsk in
-                try derivationTool.deriveTransparentAddressFromPrivateKey(tsk)
-            },
-            isValidExtendedViewingKey: { extvk in
-                try derivationTool.isValidExtendedViewingKey(extvk)
+            deriveSpendingKey: { seed, accountIndex in
+                try derivationTool.deriveUnifiedSpendingKey(seed: seed, accountIndex: accountIndex)
             },
             isValidTransparentAddress: { tAddress in
-                try derivationTool.isValidTransparentAddress(tAddress)
+                derivationTool.isValidTransparentAddress(tAddress)
             },
-            isValidShieldedAddress: { zAddress in
-                try derivationTool.isValidShieldedAddress(zAddress)
+            isValidSaplingAddress: { zAddress in
+                derivationTool.isValidSaplingAddress(zAddress)
             },
             isValidZcashAddress: { address in
-                try derivationTool.isValidTransparentAddress(address) ? true :
-                try derivationTool.isValidShieldedAddress(address) ? true : false
+                derivationTool.isValidTransparentAddress(address) ? true :
```

Comment:
The best way to do this is actually using a recipient struct and asserting on its success.  But it's not a direct use of the wrapped interface.

Reply:
Changed